### DISCUSSION TITLE
There is no selection of faces for portrait images.

### DISCUSSION THREAD

**Discussion by Impg** [lllyasviel/Fooocus#2767] [Q&A]
I use Fooocus in Colab. Generating and modifying portraits. But. No matter what I do: enlargement, variation, enhancement - Fooocus will always use one single **female** face as the only model. This doesn't happen in nature, so this solution is unnatural. Please fix the problem, bring many realistic face models into Fooocus.

**Comment by mashb1t** [comment]
This heavily depends on the used preset as well as the prompt. Please elaborate and provide more insights.

**Comment by Impg** [comment]
Hello. 
As a promt I propose to take two texts for two generations, which will allow, when comparing, to see the difference. No. 1 "He is a musician". No. 2 "She is a musician." As you can see, there is only one difference between "he" and "**she**". This illustrates the problem well. I also suggest not to use the preset for now: the artwork may distort the test result. 

Now let's generate any number of images for query  № 1. It is possible to generate several times. And each time there will be a different person as a musician. Each image will have a different character in it. And in order to get such a difference in characters I don't need **any** other generation conditions. 

Similar situation. 
Let's use query № 2. But in each image there will be one person. In different clothes, with different accessories, in different circumstances - but it is still one person, one person all the time. That's the essential difference. 

Two promts - but different results. The first result is prepared on diversity, the second on uniformization. 


The impact of uniformization spreads much wider. At the very least:

- Variation
- Face enhancement
- Changing the value of StopArt
- Upscale

In each case, only one model will be applied as the source of the changes. But the bottom line is "changes without changes". 


But can promt make a difference? Let's see.

In those circumstances where little change is allowed, promt has minimal, very little impact. It's not about changing the entourage, it's about changing the character, and specifically the female face. 

But even in cases where you can radically change the image - the situation is no better. Because there is still unification regarding the above-mentioned type of character. Let's try to express, through promt, a character of a certain ethnicity. And we will have - always one character, one face: regardless of the number of images and the number of generations. 

These results gave me the idea to ask for different models to be included in the Colab loadout. Also, maybe there should be a way to change the behavior of Fooocus to be variably change the character selection without an additional promt. 

It's more about separately enabling the generator , which is already enabled by default now for some individuals and disabled for some reason for other individuals.


Thanks .
